saged/all_label_comparison.py

```diff
@@ -0,0 +1,126 @@
+"""
+This benchmark compares the performance of different models in learning to predict all the
+classes in the dataset
+"""
+import argparse
+
+import sklearn.metrics
+import yaml
+
+from saged import utils, datasets, models
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument('dataset_config',
+                        help='The yaml formatted dataset configuration file. For more information '
+                             'about this file read the comments in the example_dataset.yml file')
+    parser.add_argument('supervised_config',
+                        help='The yaml formatted model configuration file. For more information '
+                             'about this file read the comments in the example_model.yml file')
+    parser.add_argument('out_file',
+                        help='The file to save the results to')
+    parser.add_argument('--unsupervised_config',
+                        help='The yaml formatted model configuration file for an unsupervised '
+                             'model. If omitted, the benchmark will use the original features '
+                             'for training. For more information about this file read the '
+                             'comments in the example_model.yml file',
+                        default=None)
+    parser.add_argument('--neptune_config',
+                        help='A yaml formatted file containing init information for '
+                             'neptune logging')
+    parser.add_argument('--seed',
+                        help='The random seed to be used in the experiment',
```

Comment:
Might be good to note that this seed only controls the way the data is split, not the model initialization etc. (looks like the seed for the Pytorch model is set separately in the relevant model config).

Reply:
Good point